### ISSUE TITLE
kconfig: build-in hardware RNG drivers

### ISSUE THREAD

**Issue by lucab** [coreos/bugs#2430]
As of 1772.0.0, our kernel config enables support for hardware random number generators and builds a couple of drivers as modules (`CONFIG_HW_RANDOM_VIRTIO` and `CONFIG_HW_RANDOM_TIMERIOMEM` in commonconfig). In order to speed up entropy gathering at early boot (and hopefully sidestep https://github.com/coreos/bugs/issues/2429 and similar issues), it would be convenient to build in the most common drivers.

From a quick look around, this would be my proposed list:
```
CONFIG_HW_RANDOM_VIRTIO=y
CONFIG_HW_RANDOM_CAVIUM=y
CONFIG_HW_RANDOM_VIA=y
CONFIG_HW_RANDOM_INTEL=y
CONFIG_HW_RANDOM_AMD=y
```

Two additional sidenotes:
 * `timeriomem-rng` is not available on any of the platforms we support, thus it can be safely disabled
 * in order to properly use `virtio-rng` the host needs to pass `-device virtio-rng-pci`

**Comment by glevand** [comment]
Probably want to include `HW_RANDOM_HISI` for their arm64 server chips.

**Comment by bgilbert** [comment]
`HW_RANDOM_HISI` depends on `ARCH_HISI`, which we don't set.

I agree that `timeriomem` is unlikely to be used anywhere, but it's cheap to keep building it as a module and will avoid any accidental breakage.

https://github.com/coreos/scripts/pull/813 enables `-device virtio-rng-pci` in `coreos_production_qemu.sh`.  We could do the same for kola, but it seems safer to continue to test instances that don't have special provisions for obtaining entropy.

**Comment by bgilbert** [comment]
Fixed in https://github.com/coreos/coreos-overlay/pull/3231.

**Comment by bgilbert** [comment]
This should be fixed in alpha 1786.0.1, due shortly.
